opendcs-rest-api/src/main/java/org/opendcs/odcsapi/res/AppResources.java

```diff
@@ -105,112 +134,174 @@ public Response getApp(@QueryParam("appid") Long appId)
-		throws WebAppException, DbException, SQLException
+			throws DbException
 	{
-		LOGGER.debug("Post app received app {} with id {}", app.getAppName(), app.getAppId());
-		try (DbInterface dbi = new DbInterface();
-			ApiAppDAO dao = new ApiAppDAO(dbi))
+		try (LoadingAppDAI dai = getLegacyDatabase().makeLoadingAppDAO())
 		{
-			dao.writeApp(app);
-			return ApiHttpUtil.createResponse(app);
+			dai.writeComputationApp(map(app));
+			return Response.status(HttpServletResponse.SC_OK)
+					.entity(String.format("Wrote app to database with ID: %s", app.getAppId()))
+					.build();
+		}
+		catch(DbIoException ex)
+		{
+			throw new DbException("Unable to store app", ex);
 		}
 	}
 
+	static CompAppInfo map(ApiLoadingApp app)
+	{
+		CompAppInfo ret = new CompAppInfo();
+		if (app.getAppId() != null)
+		{
+			ret.setAppId(DbKey.createDbKey(app.getAppId()));
+		}
+		ret.setAppName(app.getAppName());
+		ret.setComment(app.getComment());
+		ret.setLastModified(app.getLastModified());
+		ret.setProperties(app.getProperties());
+		ret.setManualEditApp(app.isManualEditingApp());
+		String appType = app.getProperties().getProperty("appType");
+		if (appType == null)
+		{
+			ret.setProperty("appType", app.getAppType());
+		}
+		return ret;
+	}
+
 	@DELETE
 	@Path("app")
 	@Consumes(MediaType.APPLICATION_JSON)
 	@Produces(MediaType.APPLICATION_JSON)
 	@RolesAllowed({AuthorizationCheck.ODCS_API_ADMIN, AuthorizationCheck.ODCS_API_USER})
-	public Response deletApp(@QueryParam("appid") Long appId)
-		throws WebAppException, DbException, SQLException
+	public Response deleteApp(@QueryParam("appid") Long appId)
+			throws DbException, WebAppException
 	{
-		LOGGER.debug("Delete app received request to delete app with id {}", appId);
-		
-		// Use username and password to attempt to connect to the database
-		try (DbInterface dbi = new DbInterface();
-			ApiAppDAO dao = new ApiAppDAO(dbi))
+		if (appId == null)
+		{
+			throw new WebAppException(HttpServletResponse.SC_BAD_REQUEST, "Missing required appid parameter.");
+		}
+
+		try (LoadingAppDAI dai = getLegacyDatabase().makeLoadingAppDAO())
 		{
-			dao.deleteApp(appId);
-			return ApiHttpUtil.createResponse("appId with ID " + appId + " deleted");
+			CompAppInfo app = dai.getComputationApp(DbKey.createDbKey(appId));
+			if (app == null)
+			{
+				throw new DbException(String.format("No such app with ID: %s", appId));
+			}
+			dai.deleteComputationApp(app);
+			return Response.status(HttpServletResponse.SC_OK)
+					.entity("appId with ID " + appId + " deleted").build();
+		}
+		catch (NoSuchObjectException | DbIoException | ConstraintException ex)
+		{
+			throw new DbException(String.format("No such app with ID: %s", appId), ex);
 		}
 	}
 
 	@GET
 	@Path("appstat")
 	@Produces(MediaType.APPLICATION_JSON)
 	@RolesAllowed({AuthorizationCheck.ODCS_API_ADMIN, AuthorizationCheck.ODCS_API_USER})
- 	public Response getAppStat() throws DbException
+	public Response getAppStat() throws DbException
```

Comment:
should rename from `Stat` to `Status` while we're modifying

Reply:
Renamed as suggested